PolyEngine/Engine/Src/InputWorldComponent.cpp

```diff
@@ -17,3 +17,57 @@ bool Poly::InputWorldComponent::IsPressed(const std::initializer_list<eMouseButt
 		result = result && CurrMouseButton[i];
 	return result;
 }
+
+bool Poly::InputWorldComponent::IsPressed(size_t playerID, eControllerButton button) const
+{
+	ASSERTE(IsControllerConnected(playerID), "Controller not connected!");
+	i32 joystickID = PlayerIDToJoystickID[playerID].Value();
+	return Controllers.at(joystickID).CurrButton[button];
+}
+
+bool Poly::InputWorldComponent::IsClicked(size_t playerID, eControllerButton button) const
+{
+	ASSERTE(IsControllerConnected(playerID), "Controller not connected!");
+	i32 joystickID = PlayerIDToJoystickID[playerID].Value();
+	return Controllers.at(joystickID).CurrButton[button] && !Controllers.at(playerID).PrevButton[button];
```

Comment:
It would be faster to first cache the reference to controller state:
```
const ControllerState& state = Controllers.at(joystickID);
return state.CurrButton[button] && !state.PrevButton[button];
```
That way we do not search map twice. This applies to all methods that call `at` function more than once on the same argument.